### PR TITLE
Coalesce function implementation

### PR DESCRIPTION
**Pull request by aelesbao**

### CODE REVIEW

File: sql/src/test/java/io/crate/integrationtests/ConditionalFunctionIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+public class ConditionalFunctionIntegrationTest extends SQLTransportIntegrationTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/ConditionalFunctionIntegrationTest.java]: pls remove, seems to be unused

File: sql/src/test/java/io/crate/integrationtests/ConditionalFunctionIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+public class ConditionalFunctionIntegrationTest extends SQLTransportIntegrationTest {
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/ConditionalFunctionIntegrationTest.java]: afaik there is no need for an integration test of this, so pls remove

File: sql/src/test/java/io/crate/operation/scalar/conditional/ConditionalFunctionTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+public class ConditionalFunctionTest extends AbstractScalarFunctionsTest {
```
Comment (seut) [inline sql/src/test/java/io/crate/operation/scalar/conditional/ConditionalFunctionTest.java]: pls checkout the other child implementation of AbstractScalarFunctionTest, all can be done much easier with less code ;)

File: sql/src/main/java/io/crate/operation/scalar/conditional/ConditionalFunction.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import io.crate.operation.Input;
+import io.crate.operation.scalar.ScalarFunctionModule;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public abstract class ConditionalFunction extends Scalar<Object, Object> {
```
Comment (seut) [inline sql/src/main/java/io/crate/operation/scalar/conditional/ConditionalFunction.java]: as only the function info is shared here, I'd prefer removing this abstract function. if you want one place to register all conditional function you could consider impl a ConditionalFunctionModule. but doing this for 3 function is maybe a bit overkill ;)

Comment (aelesbao) [inline sql/src/main/java/io/crate/operation/scalar/conditional/ConditionalFunction.java]: @smu I plan to reuse the Resolver pre-condition check on the other functions too. wdyt?

File: CHANGES.txt [hunk 5]
```diff
@@ -5,6 +5,8 @@ Changes for Crate
 Unreleased
 ==========
 
+ - Implemented conditional functions - ``coalesce()``.
```
Comment (seut) [inline CHANGES.txt]: as discussed, pls remove now and add this later on the story branch after all implementations are done

**Comment by seut** [comment]
otherwise lgtm

**Pull request merged**